services/src/pro/layers/postgres_layer_db.rs

```diff
@@ -331,6 +334,123 @@ where
 
         Ok(())
     }
+
+    async fn remove_collection(&self, collection: &LayerCollectionId) -> Result<()> {
+        let collection =
+            Uuid::from_str(&collection.0).map_err(|_| crate::error::Error::IdStringMustBeUuid {
+                found: collection.0.clone(),
+            })?;
+
+        if collection == INTERNAL_LAYER_DB_ROOT_COLLECTION_ID {
+            return Err(LayerDbError::CannotRemoveRootCollection.into());
+        }
+
+        let conn = self.conn_pool.get().await?;
```

Comment:
maybe open a transaction here to ensure everything is deleted.

Reply:
[c48b6ec](https://github.com/geo-engine/geoengine/pull/676/commits/c48b6ec12e37040016e367af4fbb230bf3b6923f)